### DISCUSSION TITLE
How to get the current playing state (paused, playing, stopped) via the API without websockets?

### DISCUSSION THREAD

**Discussion by l3d00m** [librespot-org/librespot-java#410] [Q&A]
Is it possible to get the current playing state of librespot via the API without websockets? If I load up [my application](https://github.com/l3d00m/MMM-LibrespotJava) initially there's no way to know whether it's currently playing, stopped, paused or whatever. Only if the next websocket event occurs (i.e. by switching tracks or toggling pause/play) I know what the current state is.

There is `POST /player/current`, but as far as I see it, it does not return information on whether the current song is paused, playing or even stopped. The only way to get that is to watch the websockets. Is that correct?

**Comment by cth-latest** [comment]
If i'm not wrong the new versions of spotify app's are using HTTP only for their entire app.

**Reply by l3d00m** [reply]
I'm not sure what you mean by this. 

My question was from the perspective of a user of the inbuilt librespot-java API, which works mostly by using websockets. I want to avoid using those websockets and use only the HTTP endpoints. But w/o websockets it's not possible to get the current playing state, whether it's stopped, paused or playing.

API docs https://github.com/librespot-org/librespot-java/tree/dev/api#api

**Reply by devgianlu** [reply]
I don't there's a way right now. You can open an issue or PR.

**Comment by tobiasguyer** [comment]
hi.. probably you are no more interestet, or librespot has that one already solved completly, but i've found quite some information on the playback with following get request (with cspot) "hm://connect-state/v1/cluster"

**Reply by cth-latest** [reply]
hm:// (Hermes Protocol) is TCP based, I don't think this would match @l3d00m needs.

EDIT: And there is no point of reviving this discussion 3 years later.
